index.js

```diff
@@ -20,7 +20,77 @@
  * @returns boolean, valid or invalid
  */
 export function validate(str, vars) {
+  let ops = [
+    "+",
+    "-",
+    "*",
+    "/",
+    "^",
+    "**",
+    "||",
+    "&&",
+    "<",
+    ">",
+    "=",
+    "!=",
+    "!~",
+    ">=",
+    "<=",
+    "xor",
+    "XOR",
+    "implies",
+    "IMPLIES",
+    "and",
+    "or",
+    "AND",
+    "OR"
+  ];
   var len = str.length;
+  var op = "";
+  for (var j = 0; j < len; j++) {
+    if (!(/[a-z]|[A-Z]|[0-9]|[.]|[-]|[(]|[)]|[\s]|[,]/.test(str[j]))) {
```

Comment:
Could you add some comments about the logic?  I am not following this.  It is testing for characters, which could either be one of your operator strings or a variable or a function name, but you append it to "op".

Reply:
Yes, I'll add comments on logic. This actually tests that its not one of the characters in order to identify operators.

Reply:
Fixed